src/main/java/seedu/address/logic/parser/DeleteStudentCommandParser.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STUDENTID;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.DeleteCommand;
+import seedu.address.logic.commands.deletestudentcommand.DeleteStudentByEmailCommand;
+import seedu.address.logic.commands.deletestudentcommand.DeleteStudentByIdCommand;
+import seedu.address.logic.commands.deletestudentcommand.DeleteStudentByIndexCommand;
+import seedu.address.logic.commands.deletestudentcommand.DeleteStudentCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.StudentId;
+
+/**
+ * Parses input arguments and creates a new DeleteCommand object
+ */
+public class DeleteStudentCommandParser implements Parser<DeleteStudentCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the
+     * DeleteCommand
+     * and returns a DeleteCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DeleteStudentCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_INDEX, PREFIX_EMAIL, PREFIX_STUDENTID);
+        boolean isIndexPresent = argMultimap.getValue(PREFIX_INDEX).isPresent();
+        boolean isEmailPresent = argMultimap.getValue(PREFIX_EMAIL).isPresent();
+        boolean isStudentIdPresent = argMultimap.getValue(PREFIX_STUDENTID).isPresent();
+        if (!isEmailPresent && !isStudentIdPresent && !isIndexPresent
```

Comment:
Would it be better to follow the design of DeleteCommand, divding the deletion by Email, Id and Index, to similarly also divide the DeleteCommandParser by their email, id and index?

Reply:
I think the benefits to abstracting out the DeleteCommandParser would be minimal since it will basically be the same code for any of the delete parameters. However, I split the DeleteCommand because there are some differences in the specifics. Nevertheless, we can explore abstracting out the parser in future iterations if we find a need for it